delegated_credentials.go

```diff
@@ -0,0 +1,250 @@
+// Copyright 2010 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tls
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/asn1"
+	"errors"
+	"fmt"
+	"time"
+)
+
+const (
+	CredentialsValidity time.Duration = 5 * time.Minute
+)
+
+var DelegatedCredentialsIdentifier = asn1.ObjectIdentifier{2, 5, 29, 99}
+
+type GetCertificate func(*ClientHelloInfo) (*Certificate, error)
+
+type DelegatedCredential struct {
+	ValidTime int64
+	PublicKey interface{}
+}
+
+func NewDelCredGetCertificateFunction(cert *Certificate) GetCertificate {
```

Comment:
Yeah, I don't really like it myself. I'm not sure the `From` is accurate. So how about simply `NewDelegatedCredentialGetCertificate`?